src/client/pps/util.go

```diff
@@ -69,6 +73,60 @@ func InputCommits(input *Input) []*pfs.Commit {
 				ID:   input.Cron.Commit,
 			})
 		}
+		if input.Github != nil {
+			result = append(result, &pfs.Commit{
+				Repo: &pfs.Repo{RepoNameFromGithubInfo(input.Github.URL, input.Github.Name)},
+				ID:   input.Github.Commit,
+			})
+		}
 	})
 	return result
 }
+
+// RepoNameFromGithubInfo returns the normalized repo name
+func RepoNameFromGithubInfo(url string, name string) string {
+	if name != "" {
+		return name
+	}
+	// We know URL looks like:
+	// "https://github.com/sjezewski/testgithook.git",
+	tokens := strings.Split(url, "/")
+	last := tokens[len(tokens)-1]
+	tokens = strings.Split(last, ".")
+	return tokens[0]
+}
```

Comment:
Again I think it's better to use `Git` instead of `Github` for the name here. Just to leave things generic. Also it's weird to me that this takes a `name` argument just to check if it's `""`, I think the caller of this function should just do that.

Reply:
I'll change the name.

This helper is used in a bunch of places in this PR. Yea, the caller could check, but this seemed DRYer. Also to note ... it seems like it could've taken the `input.Git` as the argument ... however this is also used on the worker side where there is the worker `input` type where this wouldn't work.

Reply:
Actually after reading more I don't think you actually need this helper at all. You should just implement this behavior in `setPipelineDefaults` that way you set the `Name` field once when the pipeline comes in and from that point forward you can just use it directly.